protobuf/syft_proto/execution/v1/protocol.proto

```diff
@@ -4,16 +4,16 @@ package syft_proto.execution.v1;
 option java_package = "org.openmined.syftproto.execution.v1";
 
 import "syft_proto/types/syft/v1/id.proto";
+import "syft_proto/execution/v1/role.proto";
 
 message Protocol {
     syft_proto.types.syft.v1.Id id = 1;
-    repeated string tags = 2;
-    string description = 3;
-    bool workers_resolved = 4;
-    repeated PlanAssignment plan_assignments = 5;
-}
-
-message PlanAssignment {
-    syft_proto.types.syft.v1.Id plan_id = 1;
-    syft_proto.types.syft.v1.Id worker_id = 2;
-}
+    string name = 2;
+    map<string, syft_proto.execution.v1.Role> roles = 3;
+    repeated string input_repartition = 4;
+    repeated string output_repartition = 5;
+    bool include_state = 6;
```

Comment:
Is it possible that some `Roles` will have `State` and others won't? It didn't make any difference for `Plans`, but wondering if we should move `State` into `Roles`.

Reply:
Yeah, I think we need some thinking around State for Protocols (how to specify them, etc.). But I think it should be doable to move them into Roles. I don't think a Protocol could have a global State, could it?

Reply:
Not sure. Maybe the initial model for secure model aggregation?

Reply:
Maybe. But let's do that when we have an idea of how to solve the problem?

Reply:
So resolved that we will assume there is no global state until proven otherwise. 👍